### PR TITLE
Add support for aten::slice to guard elimination.

### PR DESCRIPTION
**Pull request by resistor**

### CODE REVIEW

File: torch/csrc/jit/passes/guard_elimination.cpp [hunk 1]
```diff
@@ -247,6 +247,15 @@ struct GuardElimination {
     case aten::erf:
     case aten::erfc:
       return checkInputs(n, no_exceptions);
+    case aten::slice:
```
Comment (Krovatkin) [inline torch/csrc/jit/passes/guard_elimination.cpp]: why aren't we checking that `n->input(0)` isn't `isSummarized()`?

Comment (resistor) [reply torch/csrc/jit/passes/guard_elimination.cpp]: I don't know what that does.

**Review by eellison — COMMENTED** [review]
Can we try to enforce that every PR has a test accompanying it?

**Comment by resistor** [comment]
> Can we try to enforce that every PR has a test accompanying it?

Done

File: torch/csrc/jit/passes/guard_elimination.cpp (hunk 1 above)
Comment (resistor) [reply torch/csrc/jit/passes/guard_elimination.cpp]: Done

**Review by Krovatkin — APPROVED** [review]
:shipit:

**Pull request closed without merging**